src/workerfarm/WorkerFarm.js

```diff
@@ -25,24 +25,56 @@ class WorkerFarm extends EventEmitter {
     this.callQueue = [];
 
     this.localWorker = require(this.options.workerPath);
-    this.remoteWorker = {
-      run: this.mkhandle('run')
-    };
+    this.run = this.mkhandle('run');
 
     this.init(options);
   }
 
+  callPromiseFactory(resolve, reject, method, args) {
```

Comment:
Why not just make this construct and return a Promise rather than duplicating twice below? Even better, merge this function with `addCall` and have it do that.

Reply:
Cleaned up in ae6fa72cec76c263cd7ad4726bbb63f765eebe2d